apidef/oas/upstream.go

```diff
@@ -30,16 +35,73 @@ func (u *Upstream) ExtractTo(api *apidef.APIDefinition) {
 }
 
 type ServiceDiscovery struct {
+	// Enabled enables the Service Discovery.
+	// Old API Definition: `service_discovery.use_discovery_service`
 	Enabled             bool   `bson:"enabled" json:"enabled"` // required
-	EndpointReturnsList bool   `bson:"endpointReturnsList,omitempty" json:"endpointReturnsList,omitempty"`
-	CacheTimeout        int64  `bson:"cacheTimeout,omitempty" json:"cacheTimeout,omitempty"`
-	ParentDataPath      string `bson:"parentDataPath,omitempty" json:"parentDataPath,omitempty"`
+	// QueryEndpoint is the endpoint to call, this would probably be Consul, etcd or Eureka K/V store.
+	// Old API Definition: `service_discovery.query_endpoint`
 	QueryEndpoint       string `bson:"queryEndpoint,omitempty" json:"queryEndpoint,omitempty"`
-	TargetPath          string `bson:"targetPath,omitempty" json:"targetPath,omitempty"`
-	UseTargetList       bool   `bson:"useTargetList,omitempty" json:"useTargetList,omitempty"`
-	UseNestedQuery      bool   `bson:"useNestedQuery,omitempty" json:"useNestedQuery,omitempty"`
+	// DataPath is the namespace of the data path - where exactly in your service response the namespace can be found.
+	// For example, if your service responds with:
+	// ```
+	// {
+	//  "action": "get",
+	//  "node": {
+	//    "key": "/services/single",
+	//    "value": "http://httpbin.org:6000",
+	//    "modifiedIndex": 6,
+	//    "createdIndex": 6
+	//  }
+	// }
+	//```
+	//
+	// then your namespace would be `node.value`.
+	// Old API Definition: `service_discovery.data_path`
 	DataPath            string `bson:"dataPath,omitempty" json:"dataPath,omitempty"`
+	// UseNestedQuery enables using a combination of the data_path and parent_data_path.
+	// It is necessary when the data lives within this string-encoded JSON object.
+	// ```
+	// {
+	//  "action": "get",
+	//  "node": {
+	//    "key": "/services/single",
+	//    "value": "{"hostname": "http://httpbin.org", "port": "80"}",
+	//    "modifiedIndex": 6,
+	//    "createdIndex": 6
+	//  }
+	// }
+	//```
+	// Old API Definition: `service_discovery.use_nested_query`
+	UseNestedQuery      bool   `bson:"useNestedQuery,omitempty" json:"useNestedQuery,omitempty"`
+	// ParentDataPath is the namespace of the where to find the nested value, if `useNestedQuery` is `true`.
+	// In the above example, it would be `node.value`. You would then change the `data_path` setting to be `hostname`,
```

Comment:
here, we're using snake_notation for `data_path`, is it supposed to be `dataPath` in camelCase?